protocols/mdns/src/behaviour/iface.rs

```diff
@@ -312,14 +312,18 @@ where
                 }
                 Poll::Ready(Err(err)) if err.kind() == std::io::ErrorKind::WouldBlock => {
                     // No more bytes available on the socket to read
+                    continue;
                 }
```

Comment:
Given that we are setting `set_nonblocking(true)`, shouldn't this match arm never be hit?

https://github.com/libp2p/rust-libp2p/blob/e3db5a57813f6ba049ebd4eba26a3aaa2ff3568d/protocols/mdns/src/behaviour/socket.rs#L102-L105

Reply:
Normally, this should be caught by the `AsyncFd` implementation but afaik, there might still be the occasional `WouldBlock` error so the safest thing is to just poll again and register a waker.

Reply:
Is this thread resolved or are you maybe doing some research to better understand `WouldBlock` ?

Reply:
Oh, my bad. Thanks for clarifying @thomaseizinger! Sorry for delaying here @stormshield-frb.